service/src/main/java/uk/gov/hmcts/reform/fpl/service/CMODocmosisTemplateDataGenerationService.java

```diff
@@ -100,22 +100,36 @@
         JudgeAndLegalAdvisor judgeAndLegalAdvisor = hearingBooking.getJudgeAndLegalAdvisor();
         cmoTemplateData.putAll(commonCaseDataExtractionService.getJudgeAndLegalAdvisorData(judgeAndLegalAdvisor));
 
-        cmoTemplateData.putAll(getGroupedCMODirections(caseManagementOrder));
+        cmoTemplateData.putAll(getGroupedCMODirections(order));
 
-        cmoTemplateData.putAll(getDraftWaterMarkData());
+        if (draft) {
+            cmoTemplateData.putAll(getDraftWaterMarkData());
+        }
 
-        List<Map<String, String>> recitals = buildRecitals(caseManagementOrder.getRecitals());
+        List<Map<String, String>> recitals = buildRecitals(order.getRecitals());
         cmoTemplateData.put("recitals", recitals);
         cmoTemplateData.put("recitalsProvided", isNotEmpty(recitals));
 
-        cmoTemplateData.putAll(getSchedule(caseManagementOrder));
+        cmoTemplateData.putAll(getSchedule(order));
 
         //defaulting as 1 as we currently do not have impl for multiple CMos
         cmoTemplateData.put("caseManagementNumber", 1);
 
         return cmoTemplateData.build();
     }
 
+    private CaseManagementOrder getCaseManagementOrder(CaseData caseData) {
+        if (caseData.getCaseManagementOrder() != null) {
+            return caseData.getCaseManagementOrder();
+        }
+
+        if (caseData.getCmoToAction() != null) {
+            return caseData.getCmoToAction();
+        }
```

Comment:
[-0.5 / -1] Probably handled in progression service, but isn't cmoToAction more important? The one currently being looked at by the judge?

Reply:
This data generation is used by both the draft and action cmo events. When in the draft cmo event the draft cmo object isn't null but when it is progressed to the judge to action it is set to null so that the LA cannot see it anymore.

Reply:
yeah I think this is fine for now